docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,928 @@
+.. _discord_slash_commands:
+
+Slash commands
+===============
+
+Slash commands are one of Discord's primary methods of implementing a user-interface for bots.
+Analogous to their name, they're previewed and invoked in the Discord client by beginning your message with a forward-slash.
+
+.. image:: /images/guide/app_commands/meow_command_preview.png
+    :width: 400
+
+App commands are implemented within the :ref:`discord.app_commands <discord_app_commands>` package.
+Code examples in this page will always assume the following two imports:
+
+.. code-block:: python3
+
+    import discord
+    from discord import app_commands
+
+Setting up
+-----------
+
+To work with app commands, bots need the ``applications.commands`` scope.
+
+You can enable this scope when generating an OAuth2 URL for your bot, shown :ref:`here <discord_invite_bot>`.
+
+Defining a Tree
+++++++++++++++++
+
+First, a :class:`.app_commands.CommandTree` needs to be instaniated
+- which is a container holding all of the bot's application commands.
+
+This class contains any relevant methods for managing app commands:
+
+- :meth:`.CommandTree.add_command` to add a command
+- :meth:`.CommandTree.remove_command` to remove a command
+- :meth:`.CommandTree.get_command` to find a specific command
+- :meth:`.CommandTree.get_commands` to return all commands
+
+Preferably, this tree should be attached to the client instance to
+play well with type checkers and to allow for easy access from anywhere in the code:
+
+.. code-block:: python3
+
+    class MyClient(discord.Client):
+        def __init__(self):
+            super().__init__(intents=discord.Intents.default())
+            self.tree = app_commands.CommandTree(self)
+
+    client = MyClient()
+
+.. note::
+
+    If your project instead uses :class:`.ext.commands.Bot` as the client instance,
+    a :class:`~discord.app_commands.CommandTree` has already been defined at :attr:`.Bot.tree`,
+    so this step is largely skipped.
+
+Creating a command
+-------------------
+
+Slash commands are created by decorating an async function.
+This function is then called whenever the slash command is invoked.
+
+For example, the following code responds with "meow" on invocation:
+
+.. code-block:: python3
+
+    @client.tree.command()
+    async def meow(interaction: discord.Interaction):
+        """Meow meow meow"""
+
+        await interaction.response.send_message("meow")
+
+Functions of this pattern are called callbacks, since their execution is
+relinquished to the library to be called later.
+
+Callbacks always have one parameter, ``interaction``,
```

Comment:
I think this wording "always have one parameter" might be confuse some people into thinking callbacks can have only a single parameter and nothing more, although additional parameters are covered later in this guide as well. Maybe "always have at least one parameter" instead?